java-frontend/src/test/java/org/sonar/java/UCFGJavaVisitorTest.java

```diff
@@ -86,38 +86,222 @@ public void visit_java_file() {
+        "    return field + FIELD + x;\n" +
+        "  }\n" +
+        "}\n", expectedUCFG);
+  }
+
+  @Test
+  public void field_access_on_object() {
+    Expression.Variable aux0 = variableWithId("%0");
+    Expression.Variable aux1 = variableWithId("%1");
+    Expression.Variable aux2 = variableWithId("%2");
+    UCFG expectedUCFG = UCFGBuilder.createUCFGForMethod("A#method(LA;)Ljava/lang/String;")
+        .addBasicBlock(newBasicBlock("1")
+            .assignTo(aux0,
+                call("__id").withArgs(new Expression.FieldAccess(variableWithId("a"), variableWithId("field"))),
+                new LocationInFile(FILE_KEY, 5,11,5,18))
+            .assignTo(aux1,
+                call("__id").withArgs(new Expression.FieldAccess(new Expression.ClassName("A"), variableWithId("FIELD"))),
+                new LocationInFile(FILE_KEY, 5,21,5,28))
+          .assignTo(aux2, call("__concat").withArgs(aux0, aux1), new LocationInFile(FILE_KEY, 5,11,5,28))
+          .ret(aux2, new LocationInFile(FILE_KEY, 5,4,5,29)))
+        .build();
+    assertCodeToUCfg("class A {\n" +
+        "  String field;\n" +
+        "  static String FIELD;\n" +
+        "  String method(A a) {\n" +
+        "    return a.field + A.FIELD;\n" +
+        "  }\n" +
+        "}\n", expectedUCFG);
+  }
+
+  @Test
+  public void field_access_chain() {
+    Expression.Variable a = variableWithId("a");
+    Expression.Variable aux0 = variableWithId("%0");
+    Expression.Variable aux1 = variableWithId("%1");
+    Expression.Variable aux2 = variableWithId("%2");
+    Expression.Variable aux3 = variableWithId("%3");
+    Expression.Variable aux4 = variableWithId("%4");
+    Expression.Variable aux5 = variableWithId("%5");
+    UCFG expectedUCFG = UCFGBuilder.createUCFGForMethod("A#method(LA;)LA;")
+        .addBasicBlock(newBasicBlock("1")
+            .assignTo(aux0,
+                call("__id").withArgs(new Expression.FieldAccess(variableWithId("foo"))),
+                new LocationInFile(FILE_KEY, 7,4,7,7))
+            .assignTo(aux1,
+                call("__id").withArgs(new Expression.FieldAccess(aux0, variableWithId("foo"))),
+                new LocationInFile(FILE_KEY, 7,4,7,11))
+            .assignTo(aux2,
+                call("__id").withArgs(new Expression.FieldAccess(aux1, variableWithId("bar"))),
+                new LocationInFile(FILE_KEY, 7,4,7,15))
+            .assignTo(aux3,
+                call("__arrayGet").withArgs(aux2),
+                new LocationInFile(FILE_KEY, 7,4,7,18))
+            .assignTo(aux4,
+                call("__id").withArgs(new Expression.FieldAccess(a, variableWithId("foo"))),
+                new LocationInFile(FILE_KEY, 7,25,7,30))
+            .assignTo(aux5,
+                call("__id").withArgs(new Expression.FieldAccess(aux4, variableWithId("baw"))),
+                new LocationInFile(FILE_KEY, 7,25,7,34))
+            .assignTo(new Expression.FieldAccess(aux3, variableWithId("baz")),
```

Comment:
This order looks strange to me. Why is this occurring after evaluating the right hand side (`RHS`)? I would have expected to have the `LHS` fully completed before evaluating the RHS.